tests/e2e/macvtap_test.go

```diff
@@ -0,0 +1,301 @@
+				daemonSetNamePrefix := "macvtap-cni"
+				pods, _ := clientset.CoreV1().Pods(namespace).List(metav1.ListOptions{})
+				nodes, _ := clientset.CoreV1().Nodes().List(metav1.ListOptions{})
+				Expect(len(pods.Items)).Should(BeNumerically(">", 0))
+				Expect(len(nodes.Items)).Should(BeNumerically(">", 0))
+
+				networkStatus := FilterPods(pods.Items, func(pod v1.Pod) bool {
+					return strings.HasPrefix(pod.Name, daemonSetNamePrefix)
+				})
+				Expect(len(networkStatus)).To(Equal(len(nodes.Items)))
+			})
+
+			It("THEN a macvtap custom network resource is exposed", func() {
+				quantity := 50
+
+				expectedResourceName := v1.ResourceName(getMacvtapResourceName(lowerDevice))
+				waitForResourceAvailability(Timeout, getMacvtapResourceName(lowerDevice), metav1.ListOptions{})
+
+				nodes, _ := clientset.CoreV1().Nodes().List(metav1.ListOptions{})
+				for _, node := range nodes.Items {
+					expectedQuantity, err := resource.ParseQuantity(strconv.Itoa(quantity))
+					Expect(err).NotTo(HaveOccurred())
+
+					// confirm capacity is OK
+					macvtapCapacity := node.Status.Capacity[v1.ResourceName(expectedResourceName)]
+					Expect(macvtapCapacity).To(Equal(expectedQuantity))
+
+					// confirm allocatable is OK
+					macvtapAllocatable := node.Status.Allocatable[v1.ResourceName(expectedResourceName)]
+					Expect(macvtapAllocatable).To(Equal(expectedQuantity))
+				}
+			})
+		})
+	})
+
+	Describe("macvtap resource creation", func() {
+
+		Context("WHEN a macvtap interface is configured as a secondary interface", func() {
+			macvtapIfaceName := "macvtap0"
+			postUrl := "/apis/k8s.cni.cncf.io/v1/namespaces/%s/network-attachment-definitions/%s"
+			nad := `
+					{
+						"apiVersion":"k8s.cni.cncf.io/v1",
+						"kind":"NetworkAttachmentDefinition",
+						"metadata": {
+							"name":"%s",
+							"namespace":"%s",
+							"annotations": {
+								"k8s.v1.cni.cncf.io/resourceName": "%s"
+							}
+						},
+						"spec":{
+							"config":"{ \"cniVersion\": \"0.3.1\", \"name\": \"%s\", \"type\": \"macvtap\"}"
+						}
+					}
+				`
+
+			BeforeEach(func() {
+				clientset.RESTClient().
+					Post().
+					RequestURI(fmt.Sprintf(postUrl, namespace, macvtapIfaceName)).
+					Body([]byte(fmt.Sprintf(nad, macvtapIfaceName, namespace, getMacvtapResourceName(lowerDevice), macvtapIfaceName))).
+					Do()
+			})
+
+			AfterEach(func() {
+				clientset.RESTClient().
+					Delete().
+					RequestURI(fmt.Sprintf(postUrl, namespace, macvtapIfaceName)).
+					Do()
+			})
+
+			Context("WHEN a pod requests aforementioned macvtap resource", func() {
+				podName := "megapod"
+				containerName := "tinycontainer"
+
+				container := v1.Container{
+					Name:    containerName,
+					Image:   "alpine",
+					Command: []string{"/bin/sh", "-c", "sleep 999999"},
+					Resources: v1.ResourceRequirements{
+						Limits: getMacvtapResourceRequest(lowerDevice, 1),
+					},
+				}
+
+				pod1 := &v1.Pod{
+					TypeMeta: metav1.TypeMeta{
```

Comment:
I believe this is not needed. The TypeMeta is already included in `Pod`.

Reply:
Done.